docs/technical/mock-classifier-technical-architecture.md

```diff
@@ -0,0 +1,110 @@
+
+| Ministry Name                                                | Label       |
+| :----------------------------------------------------------- | ----------- |
+| [Ministry of Justice](https://www.just.ee/en)                | justice     |
+| [Ministry of Education and Research](https://www.hm.ee/en)   | education   |
+| [Ministry of Defence](https://www.kaitseministeerium.ee//en) | defence     |
+| [Ministry of the Environment](https://www.envir.ee/en)       | environment |
+| [Ministry of Cultural Affairs](https://www.kul.ee/en)        | cultural    |
+| [Ministry of Rural Affairs](https://www.agri.ee/en)          | rural       |
+| [Ministry of Economic Affairs and Communications](https://www.mkm.ee/en) | economic    |
+| [Ministry of Finance](https://www.rahandusministeerium.ee/en) | finance     |
+| [Ministry of Social Affairs](https://www.sm.ee/en)           | social      |
+| [Ministry of the Interior](https://www.siseministeerium.ee/en) | interior    |
+| [Ministry of Foreign Affairs](https://vm.ee/en)              | foreign     |
+
+## Natural Language Resolutions
+
+The Classifier will be hard-coded to support the following natural language resolutions for testing and demo purposes.
+
+| Input Text (all in English)                                  | Returns     |
+| ------------------------------------------------------------ | ----------- |
+| I want to register my child at school                        | education   |
+| How do I file my annual tax information                      | economic    |
+| I have a question about the Estonian pension system          | economic    |
+| How do I get to Lahemaa park?                                | environment |
+| How do I arrange for my COVID-19 booster vaccination         | social      |
+| I wish to understand what benefits my family are entitled to | social      |
+
+Additional resolutions may be added during the course of implementation as required.
+
+These resolutions will be hard coded into the mock classifier and will not require any external configuration or database.
+
+> This means that if new input text needs to be added, the source code will need to be modified. This is acceptable given that this is a throw-away transient service until the real classifier is ready.
+
+## Token Resolutions
+
+The Classifier will accept special tokens anywhere in the input text which will inform the Classifier which ministry to resolve.
+
+The token format will be `{{token}}` where token is one of labels listed in the Ministries section.
+
+Here are some examples:
+
+- An input text of `Please return the {{social}} minsitry` will return `social`
+- An input text of `{{environment}}` will return `environment`
+- An input text of `{{rural}} please` will return `rural`
+
+Spaces are not permitted so an input text of `Please return {{ environment}}` would return an error code.
+
+## API Design
+
+The Classifier will be a REST api with just a single endpoint which is `/resolution`.
```

Comment:
What about adding `/random` endpoint as well, using all of the labels outlined in the Ministries section as an input? Although the impact would be minimal, it would still have some effect on benchmarking on high volumes.

Reply:
You mean /random would return a random ministry? Maybe it would be easier to use the same endpoint, but use {{random}} in the input message?

Reply:
Yes I think it is cleaner to use the existing token method for random. I'll add that into the doc

Reply:
Updated in commit 1a2474973d9c53c512d6a2b502a79f911521330b